Library/Homebrew/formula.rb

```diff
@@ -4237,12 +4237,10 @@ def pour_bottle?(only_if: nil, &block)
     # @see https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae
     # @see DeprecateDisable::FORMULA_DEPRECATE_DISABLE_REASONS
     # @api public
-    def deprecate!(date:, because:)
+    def deprecate!(date:, because: nil)
```

Comment:
Why is the `because` no longer needed?

Reply:
It is shared with `disable!`. The goal is to allow the DSL from https://github.com/Homebrew/brew/pull/18551#issuecomment-2412100504, e.g.:
```rb
disable! date: "2025-10-01", because: :does_not_build
deprecate! date: "2024-10-01"
```

And then `brew info` shows:
> Deprecated because it does not build! It will be disabled on 2025-10-01.

```json
      "deprecated": true,
      "deprecation_date": "2024-10-01",
      "deprecation_reason": "does_not_build",
      "disabled": false,
      "disable_date": "2025-10-01",
      "disable_reason": null,
```

---

The audit changes are to make sure a missing `because:` on deprecate must be accompanied by a `because:` on disable.

Reply:
Gotcha, thanks, seems desirable 👍🏻.